evaluator/evaluator.py

```diff
@@ -262,6 +291,7 @@ def run(self):
         for sig in signals:
             loop.add_signal_handler(
                 sig, lambda sig=sig: loop.create_task(terminate(sig, loop)))
+        executor = BoundedExecutor(MAX_QUEUE_SIZE, max_workers=WORKER_THREADS)
 
         # pylint: disable=too-many-branches
         def process_message(message):
```

Comment:
Is it necessary to have nested function here? It's easy to write nested function, but hard to understand. How you can simply see, what is output and input? Also it increases parent function length what is also bad. Try to keep functions as short, as easy to read, as simple and as flat as possible. I, as lazy and slow programmer, will be grateful for that :smiley:.

Reply:
I agree but let's address this re-factoring in separate PR.

Reply:
> But this is out of the scope of this PR, I guess we could create a new one task to refactor things like this?

Reply:
Yes, this is what I meant.